### PR TITLE
ReaderRolling: Guard against races between scheduled tasks and CloseDocument

### LINKED ISSUE #10932: Timesync and wifi issue
* KOReader version: 2023.08
* Device: Kindle Voyage

#### Issue

First, I would like to thank the developers for their fantastic work.

I have two issues. 

The first issue is that timesync is not working. Time is always lagging a few minutes. I don't know if it's supposed to update automatically. And yes, I am connected to wifi. In any case, I also apply "Synchronize time" from the menu, and I get a message that the time was updated but the updated time is still wrong.

The second issue I have is that toggling wifi crashes Koreader once every 2-3 times.

#### Steps to reproduce

Apply "Synchronize time" from the menu, the time is updated but it is wrong (lagging a few minutes).

Toggling wifi crashes Koreader once every 2-3 times.

##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to [Menu] → Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file (zipped if necessary) on GitHub by dragging and dropping it onto this textbox.

If your issue doesn't directly concern a Lua crash, we'll quite likely need you to reproduce the issue with *verbose* debug logging enabled before providing the logs to us.
To do so, from the file manager, go to [Tools] → More tools → Developer options, and tick both `Enable debug logging` and `Enable verbose debug logging`.
You'll need to restart KOReader after toggling these on.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
[crash.log](https://github.com/koreader/koreader/files/12702684/crash.log)

  <summary>crash.log</summary>

```
<Paste crash.log content here>
```
</details>

### PR DESCRIPTION
**Pull request by NiLuJe**
Fix #10932

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10934)

### CODE REVIEW

**Comment by NiLuJe** [comment]
(Ping @poire-z for when he gets back ;)).

File: frontend/apps/reader/modules/readerrolling.lua [hunk 1]
```diff
@@ -1087,9 +1102,7 @@ function ReaderRolling:updatePos(force)
     -- Allow for the new rendering to be shown before possibly showing
     -- the "Styles have changed..." ConfirmBox so the user can decide
     -- if it is really needed
-    UIManager:scheduleIn(0.1, function ()
-        self:onCheckDomStyleCoherence()
-    end)
+    UIManager:scheduleIn(0.1, self.onCheckDomStyleCoherence, self)
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerrolling.lua:1105]: I prefer the readability of the one with the function() wrapper - and for consistency with the others that may do just a bit more, like in your next chunk.

**Review by poire-z — APPROVED** [review]
Nothing against.

File: frontend/apps/reader/modules/readerrolling.lua (hunk 1 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerrolling.lua:1105]: It usually depends on my mood, but avoiding a closure here means we can easily unschedule it without storing a reference to the scheduled closure, so it's very much on purpose there.

(I'm not necessarily a fan of doing the guarding *inside* the closure like some of them do, for instance).

File: frontend/apps/reader/modules/readerrolling.lua [hunk 2]
```diff
@@ -801,7 +815,7 @@ end
 
 function ReaderRolling:onGotoXPointer(xp, marker_xp)
     if self.mark_func then
-        -- unschedule previous marker as it's no more accurate
+        -- unschedule previous marker as it's no longer accurate
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerrolling.lua:818]: If you're touching random comments touch 'em up all the way. :-P
```suggestion
        -- Unschedule previous marker as it's no longer accurate.
```

**Pull request merged**